policy-controller/k8s/status/src/lib.rs

```diff
@@ -0,0 +1,121 @@
+                    .map(|parent| {
+                        // Is this parent in the list of parents which accept
+                        // the route?
+                        let accepted = accepted_routes
+                            .get(&name)
+                            .into_iter()
+                            .flatten()
+                            .any(|accepting_parent| accepting_parent == &parent.name);
+                        let condition = if accepted {
+                            k8s::Condition {
+                                last_transition_time: k8s::Time(Utc::now()),
+                                message: "".to_string(),
+                                observed_generation: None,
+                                reason: "Accepted".to_string(),
+                                status: "True".to_string(),
+                                type_: "Accepted".to_string(),
+                            }
+                        } else {
+                            k8s::Condition {
+                                last_transition_time: k8s::Time(Utc::now()),
+                                message: "".to_string(),
+                                observed_generation: None,
+                                reason: "NoMatchingParent".to_string(),
+                                status: "False".to_string(),
+                                type_: "Accepted".to_string(),
+                            }
+                        };
+                        gateway::RouteParentStatus {
+                            parent_ref: gateway::ParentReference {
+                                group: Some("policy.linkerd.io".to_string()),
+                                kind: Some("Server".to_string()),
+                                namespace: Some(namespace.clone()),
+                                name: parent.name.clone(),
+                                section_name: None,
+                                port: None,
+                            },
+                            controller_name: "policy.linkerd.io/status-controller".to_string(),
+                            conditions: vec![condition],
+                        }
+                    })
+                    .collect();
+                let status = gateway::HttpRouteStatus {
+                    inner: gateway::RouteStatus {
+                        parents: parent_statuses,
+                    },
+                };
+                let patch = serde_json::json!({
+                    "apiVersion": "policy.linkerd.io/v1alpha1",
+                    "kind": "HTTPRoute",
+                    "name": name,
+                    "status": status,
+                });
+                if let Err(error) = api
+                    .patch_status(&name, &patch_params, &k8s::Patch::Merge(patch))
+                    .await
+                {
+                    tracing::error!(%error, "Failed to patch HTTPRoute");
```

Comment:
should this also include the name and namespace of the HTTPRoute that we failed to patch?